lib/graphql/tracing/data_dog_tracing.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module GraphQL
+  module Tracing
+    class DataDogTracing < PlatformTracing
+      self.platform_keys = {
+        'lex' => 'lex.graphql',
+        'parse' => 'parse.graphql',
+        'validate' => 'validate.graphql',
+        'analyze_query' => 'analyze.graphql',
+        'analyze_multiplex' => 'analyze.graphql',
+        'execute_multiplex' => 'execute.graphql',
+        'execute_query' => 'execute.graphql',
+        'execute_query_lazy' => 'execute.graphql',
+      }
+
+      def platform_trace(platform_key, key, data)
+        pin = Datadog::Pin.get_from(self)
+        unless pin
+          pin = Datadog::Pin.new('graphql-ruby', app: 'graphql-ruby')
+          pin.onto(self)
+        end
+
+        pin.tracer.trace(platform_key, service: pin.service) do |span|
+          if key == 'execute_multiplex'
+            span.resource = data[:multiplex].queries.map(&:selected_operation_name).join(', ')
+          end
+
+          if key == 'execute_query'
+            span.set_tag(:selected_operation_name, data[:query].selected_operation_name)
+            span.set_tag(:selected_operation_type, data[:query].selected_operation.operation_type)
+            span.set_tag(:query_string, data[:query].query_string)
+            span.set_tag(:variables, data[:query].variables.to_h)
+          end
+          yield
+        end
+      end
+
+      def platform_field_key(type, field)
+        "#{type.name}.#{field.name}"
```

Comment:
It might be helpful to have the query name here as well -- that makes it easier to find a specific query inside DataDog's UI.

Reply:
There is strange aggregation strategy in datadog panel. If that field on the top level is different, datadog redirects to the most popular query name when You click on the service name :(